geocode-api/src/main/java/org/gbif/geocode/api/service/GadmService.java

```diff
@@ -0,0 +1,28 @@
+package org.gbif.geocode.api.service;
+
+import org.gbif.api.model.common.paging.Pageable;
+import org.gbif.api.model.common.paging.PagingResponse;
+import org.gbif.geocode.api.model.GadmRegion;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+public interface GadmService {
+
+  Collection<GadmRegion> listLevel0(@Nullable String query);
+
+  Collection<GadmRegion> listLevel1(@Nullable String query, String gid0);
+
+  Collection<GadmRegion> listLevel2(@Nullable String query, String gid0, String gid1);
+
+  Collection<GadmRegion> listLevel3(@Nullable String query, String gid0, String gid1, String gid2);
+
+  PagingResponse<GadmRegion> search(@Nullable String query,
+                                    @Max(value = 2, message = "Only levels 0, 1 and 2 are supported")
```

Comment:
Shouldn't level 3 be supported?

Reply:
I added it because that is the data we have in the table gadm3, I can just remove it by now

Reply:
We have levels 0, 1, 2 and 3 in the gadm3 table...?

```
select <some> from gadm3 where gid_3 is not null limit 1;
fid               │ 245200
uid               │ 246045
gid_0             │ PHL
id_0              │ 179
name_0            │ Philippines
gid_1             │ PHL.51_1
id_1              │ 51
name_1            │ Negros Occidental
varname_1         │ ␀
nl_name_1         │ ␀
hasc_1            │ PH.ND
cc_1              │ 645
type_1            │ Lalawigan|Probinsya
engtype_1         │ Province
validfr_1         │ Unknown
validto_1         │ Present
remarks_1         │ ␀
gid_2             │ PHL.51.15_1
id_2              │ 1045
name_2            │ Kabankalan City
varname_2         │ ␀
nl_name_2         │ ␀
hasc_2            │ PH.ND.KB
cc_2              │ 64515
type_2            │ Lungsod|Siyudad
engtype_2         │ City
validfr_2         │ 19970802
validto_2         │ Present
remarks_2         │ ␀
gid_3             │ PHL.51.15.14_1
id_3              │ 27536
name_3            │ Camugao
varname_3         │ ␀
nl_name_3         │ ␀
hasc_3            │ ␀
cc_3              │ ␀
type_3            │ Barangay
engtype_3         │ Village
validfr_3         │ Unknown
validto_3         │ Present
remarks_3         │ ␀
geom              │ 
fulltext_search_0 │ 'philippin':2 'phl':1
fulltext_search_1 │ 'negro':2 'occident':3 'phl.51_1':1
fulltext_search_2 │ 'citi':3 'kabankalan':2 'phl.51.15_1':1
fulltext_search_3 │ 'camugao':2 'phl.51.15.14_1':1
```

Reply:
ah sorry, I thought you were pointing to the listLevelX methods, I will change the search method

Reply:
Level3 added
https://github.com/gbif/geocode/pull/13/commits/dd4b7a7a9872b0138b8ee6ddb7599ae1a8c858ae